lib/puppetx/simp/compliance_mapper.rb

```diff
@@ -374,135 +374,181 @@ def import(filename, data)
         def list_puppet_params(profile_list)
           retval = {}
 
+          # Potential matches prior to confinement
+          specifications = []
+
           profile_list.reverse.each do |profile|
-            if @profile_list.key?(profile)
-              info = @profile_list[profile]
-
-              @check_list.each do |check, spec|
-                specification = Marshal.load(Marshal.dump(spec))
-                continue = true
-
-                if (specification["type"] == "puppet") || (specification["type"] == "puppet-class-parameter")
-                  contain = false
-                  if info.key?("checks")
-                    if info["checks"].key?(check)
-                      if info["checks"][check] == true
-                        contain = true
-                      else
-                        contain = false
-                        continue = false
-                      end
-                    end
+            unless @profile_list.key?(profile)
+              @callback.debug(%{SKIP: Profile '#{profile}' not in '#{@profile_list.keys.join("', '")}'})
+              next
+            end
+
+            info = @profile_list[profile]
+
+            @check_list.each do |check, spec|
+              specification = Marshal.load(Marshal.dump(spec))
+
+              # Skip unless this item applies to puppet
+              unless (specification['type'] == 'puppet') || (specification['type'] == 'puppet-class-parameter')
+                @callback.debug("SKIP: '#{check}' is not a puppet parameter")
+                next
+              end
+
+              # Skip unless we actually have a parameter setting
+              unless specification.key?('settings')
+                @callback.debug("SKIP: '#{check}' does not have any settings")
+                next
+              end
+
+              unless specification['settings'].key?('parameter')
+                @callback.debug("SKIP: '#{check}' does not have a parameter specified")
+                next
+              end
+
+              # A parameter with a setting but without a value is invalid
+              unless specification['settings'].key?('value')
+                if specification['telemetry'] && specification['telemetry'].first
+                  location = specification['telemetry'].first['filename']
+                end
+
+                location ||= 'unknown'
```

Comment:
Ruby scope can be funny.  I would set `location= 'unknown'` prior to the if block.

Reply:
Good call